### PR TITLE
Bso 2 draw efif

### PR DESCRIPTION
**Pull request by efifogel**
## Summary of Changes

Exposed required member functions of the GeneralPolygonWithHoles_2 concept, and fixed the drawing of objects the type of which is model of the concept, i.e., the template General_polygon_with_holes_2. 

## Release Management

* Affected package(s): Polygon, Boolean_set_operations_2
* Issue(s) solved (if any): NA
* Feature/Small Feature (if any): [Features/Small Features/Enhancing General Polygons with Holes](https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Enhancing_General_Polygons_with_Holes)
* Link to compiled documentation [GeneralPolygonWithHoles_2](https://www.cs.tau.ac.il/~efif/doc_output7/Polygon/classGeneralPolygonWithHoles__2.html)
* License and copyright ownership: TAU

### CODE REVIEW

File: Boolean_set_operations_2/include/CGAL/draw_polygon_set_2.h [hunk 1]
```diff
@@ -17,71 +17,197 @@
...
+  virtual void add_elements() {
+    clear();
+
+    std::vector<Pwh> pwhs;
+    m_ps.polygons_with_holes(std::back_inserter(pwhs));
+    for (const auto& pwh : pwhs) add_elements(pwh);
+  }
+
+  /*/ Obtain the pixel ratio.
+   */
```
Comment (albert-github) [inline Boolean_set_operations_2/include/CGAL/draw_polygon_set_2.h:101]: `/*/` or `/*!`?

File: Installation/CHANGES.md [hunk 2]
```diff
@@ -129,6 +129,12 @@ extracted from labeled images.
     straight skeletons of polygons with holes. The output is a closed, combinatorially 2-manifold
     surface triangle mesh.
 
+### [2D Regularized Boolean Set Operations](https://doc.cgal.org/5.6/Manual/packages.html#PkgBooleanSetOperations2)
+-   Exposed all required member functions of the GeneralPolygonWithHoles_2 concept (e.g., `clear_outer_boundary()`, `clear_holes()`, and `clear()`).
```
Comment (albert-github) [inline Installation/CHANGES.md:133]: GeneralPolygonWithHoles_2 between backticks?

File: Polygon/include/CGAL/draw_polygon_with_holes_2.h [hunk 3]
```diff
@@ -23,103 +23,148 @@
...
+ * \ingroup PkgDrawPolygonWithHoles2
+ *
+ * opens a new window and draws `aph`, an instance of the
+ * `CGAL::Polygon_with_holes_2` class. A call to this function is blocking, that
+ * is the program continues as soon as the user closes the window. This function
+ * requires `CGAL_Qt5`, and is only available if the macro
+ * `CGAL_USE_BASIC_VIEWER` is defined.  Linking with the cmake target
+ * `CGAL::CGAL_Basic_viewer` will link with `CGAL_Qt5` and add the definition
+ * `CGAL_USE_BASIC_VIEWER`.  \tparam PH an instance of the
+ * `CGAL::Polygon_with_holes_2` class.  \param aph the polygon with holes to
```
Comment (albert-github) [inline Polygon/include/CGAL/draw_polygon_with_holes_2.h:35]: `\tparam` and `\param in running text.

File: Polygon/include/CGAL/draw_polygon_with_holes_2.h [hunk 4]
```diff
@@ -23,103 +23,148 @@
...
-
   /// Construct the viewer.
   /// @param ap2 the polygon to view
   /// @param title the title of the window
-  SimplePolygonWithHoles2ViewerQt(QWidget* parent, const P2& ap2,
-                                  const char* title="Basic Polygon_with_holes_2 Viewer") :
-    // First draw: vertices; edges, faces; multi-color; no inverse normal
-    Base(parent, title, true, true, true, false, false)
+  Pwh_2_basic_viewer_qt(QWidget* parent, const Pwh& pwh,
+                        const char* title = "Basic Polygon_with_holes_2 Viewer") :
```
Comment (albert-github) [inline Polygon/include/CGAL/draw_polygon_with_holes_2.h:66]: Miss match in names (`ap2` and `pwh`)
`parent` not documented

File: Polygon/include/CGAL/draw_polygon_with_holes_2.h [hunk 5]
```diff
@@ -23,103 +23,148 @@
...
+    auto ratio = camera()->pixelGLRatio(p);
+    m_pixel_ratio = ratio;
+    add_elements();
   }
 
-  void compute_elements(const P2& p2)
-  {
-    if (p2.outer_boundary().is_empty()) return;
+  /*/ Obtain the pixel ratio.
+   */
```
Comment (albert-github) [inline Polygon/include/CGAL/draw_polygon_with_holes_2.h:93]: `/*/` -> `/*!`?

**Comment by efifogel** [comment]
All good comments; I fixed them.
Thanks.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
This PR has been merged. Does it mean you have accepted the small feature https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/Enhancing_General_Polygons_with_Holes, @sloriot?

**Comment by efifogel** [comment]
I played it on the safe side.
The functionality was there ever since; it was simply missing in the doc.
(I refer to the 1st item in the summary.)
The second item is indeed a bug fix.
We (you) can safely accept it whatsoever (in my opinion).
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Pull request merged**